### PR TITLE
test(mempool):  Add two `Update` benchmarks

### PR DESCRIPTION
**Pull request by hvanz**
Add new `BenchmarkUpdate` and `BenchmarkUpdateAndRecheck`.
Also fix `BenchmarkCheckDuplicateTx`.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: types/tx.go [hunk 1]
```diff
@@ -91,12 +91,6 @@ func (txs Txs) hashList() [][]byte {
 // Txs is a slice of transactions. Sorting a Txs value orders the transactions
 // lexicographically.
 
-func (txs Txs) Len() int      { return len(txs) }
```
Comment (melekes) [inline types/tx.go:94]: Go API break => we can't backport to v0.38.x

**Review by melekes — APPROVED** [review]

File: types/tx.go (hunk 1 above)
Comment (jmalicevic) [reply types/tx.go:94]: Do we need this field only for this benchmark? If so, can we extract it somehow differently to not break APIs jsut for the sake of testing. If we are using it somewhere else in the mempool lanes work then this function should go together in a PR with where it is useful in production, not here.

Comment (melekes) [reply types/tx.go:94]: Or we can just leave it there for v0.38.x

Comment (hvanz) [reply types/tx.go:94]: We can also mark these functions as deprecated (even in v0.38.x).

Comment (jmalicevic) [reply types/tx.go:94]: Then we can leave them in 0.38.x, mark them as deprecated in v1.x and remove them later on.

**Review by jmalicevic — APPROVED** [review]

**Comment by hvanz** [comment]
@Mergifyio backport v0.38.x

**Pull request merged**